File: POO_Php_Python/Game/class/Coin.php
<?php 

namespace App\Game\class;

use App\Game\interface\Randomize;

class Coin implements Randomize {
    //use App\Game\Loggable;
    private $flips;

    public function __construct($flips) {
        $this -> flips = $flips;
    }

    private function flip() {
        return rand(0, 1);
    }

    public function roll() {
        return $this -> flipRecursive($this -> flips);
    }

    private function flipRecursive($remainingThrows) {
        if ($remainingThrows == 0) {
            return 1;
        }
        return $this -> flip() && $this -> flipRecursive($remainingThrows - 1);
    }

    /*
    public function getMaxValue() {
        return $this->throws;
    }
    */

}

?>

File: POO_PYTHON/Coin.py
import random

class Coin:
    def __init__(self, flips):
        self.flips = flips

    def roll(self):
        return self._flip(self.flips)

    def _flip(self, flips_left):
        if flips_left == 0:
            return 1 if random.randint(1, 2) == 1 else 2
        else:
            flip = random.randint(1, 2)
            if flip == 1:
                return self._flip(flips_left - 1)
            else:
                return 2
